lua/nvim-treesitter/configs.lua

```diff
@@ -1,88 +1,327 @@
-  },
-  go = {
+  }
+}
+
+parsers.go = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-go",
     files = { "src/parser.c" },
-  },
-  ruby = {
+  }
+}
+
+parsers.ruby = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-ruby",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  bash = {
+  }
+}
+
+parsers.bash = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-bash",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  php = {
+  }
+}
+
+parsers.php = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-php",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  java = {
+  }
+}
+
+parsers.java = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-java",
     files = { "src/parser.c" },
-  },
-  html = {
+  }
+}
+
+parsers.html = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-html",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  julia = {
+  }
+}
+
+parsers.julia = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-julia",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  json = {
+  }
+}
+
+parsers.json = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-json",
     files = { "src/parser.c" },
-  },
-  css = {
+  }
+}
+
+parsers.css = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-css",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  ocaml = {
+  }
+}
+
+parsers.ocaml = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-ocaml",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  swift = {
+  }
+}
+
+parsers.swift = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-swift",
     files = { "src/parser.c" },
-  },
-  csharp = {
+  }
+}
+
+parsers.csharp = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-c-sharp",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  typescript = {
+  }
+}
+
+parsers.typescript = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-typescript",
     files = { "src/parser.c", "src/scanner.c" },
     location = "tree-sitter-typescript/typescript"
-  },
-  tsx = {
+  }
+}
+
+parsers.tsx = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-typescript",
     files = { "src/parser.c", "src/scanner.c" },
     location = "tree-sitter-tsx/tsx"
   }
 }
 
+-- @enable can be true or false
+-- @disable is a list of languages, only relevant if enable is true
+-- @keymaps list of user mappings for a given module if relevant
+-- @is_enabled function which, given a ft, will return true if the ft works on the module
+local config = {
+  highlight = {
+    enable = false,
+    disable = {},
+    is_enabled = function(ft)
+      return queries.get_query(ft, 'highlights') ~= nil
+    end
```

Comment:
Maybe `is_enabled` is not the clearest name here. Something like `is_supported` could be clearer.